public/src/components/teacher/GroupResults.tsx

```diff
@@ -0,0 +1,119 @@
+import * as React from "react";
+import { IAssignment, ICourse, IStudentSubmission, IUser, IUserCourseWithUser, IGroupCourseWithGroup } from "../../models";
+
+import { DynamicTable, Row, Search, StudentLab } from "../../components";
+
+interface IResultsProp {
+    course: ICourse;
+    groups: IGroupCourseWithGroup[];
+    labs: IAssignment[];
+    onApproveClick: (submissionID:number) => void;
+}
+interface IResultsState {
+    assignment?: IStudentSubmission;
+    groups: IGroupCourseWithGroup[];
+}
+class GroupResults extends React.Component<IResultsProp, IResultsState> {
+    constructor(props: IResultsProp) {
+        super(props);
+
+        
+        if (this.props.groups[0] && this.props.groups[0].course.assignments[0]) {
```

Comment:
Are all these `0` indices safe? That is, could it happen that `groups[0]` is actually empty (`nil`)? I'm asking because elsewhere (e.g. line 40) you are checking `this.props.groups.length > 0`.

Reply:
Perhaps replace `this.props.groups[0]` with `currentGroup` or just `group` (following the same logic as for the proposed `student` variable in Results.tsx.)

Reply:
As I have understood typescript, if there is not a question mark when declaring the variable it can not be null, since I'm not used to this I have an old habit of not trusting it. 
I will double check this, and implement the proposed changes.

Reply:
You are correct, still need to check for length of array, since it is possible to pass down a empty array. So I will correct this.